src/code/z_player_lib.c

```diff
@@ -1187,21 +2087,275 @@ void Player_DrawCouplesMask(PlayState* play, Player* player) {
     AnimatedMat_DrawOpa(play, Lib_SegmentedToVirtual(&object_mask_meoto_Matanimheader_001CD8));
 }
 
-#pragma GLOBAL_ASM("asm/non_matchings/code/z_player_lib/func_80127594.s")
+void Player_DrawCircusLeadersMask(PlayState* play, Actor* actor) {
+    static Vec3f D_801C0BA8 = { 0.0f, 0.0f, 0.0f };
+    static Vec3f D_801C0BB4 = { 0.0f, 0.0f, 0.0f };
+    Gfx* gfx;
+    s32 i;
+
+    OPEN_DISPS(play->state.gfxCtx);
+
+    gfx = POLY_XLU_DISP;
+
+    for (i = 0; i < ARRAY_COUNT(D_801C0B90); i++) {
+        f32 temp_f22 = (D_801F59C8[i] / 400.0f) * 0.1f;
+
+        Matrix_MultVec3f(&D_801C0B90[i], &D_801F59B0[i]);
+
+        if (1) {}
+
+        D_801F59B0[i].y += -10.0f * temp_f22;
+
+        if (D_801F59C8[i] < 0x190) {
+            f32 phi_f20;
+
+            if (temp_f22 > 0.05f) {
+                phi_f20 = 0.05f;
+            } else {
+                phi_f20 = temp_f22;
+            }
+
+            Matrix_Push();
+            Matrix_Translate(D_801F59B0[i].x, D_801F59B0[i].y, D_801F59B0[i].z, 0);
+            Matrix_Scale(phi_f20, temp_f22, phi_f20, 1);
+
+            gSPMatrix(&gfx[0], Matrix_NewMtx(play->state.gfxCtx), G_MTX_PUSH | G_MTX_LOAD | G_MTX_MODELVIEW);
+            gSPSegment(&gfx[1], 0x08, gSegments[SEGMENT_NUMBER(gEffBubble1Tex)] + SEGMENT_OFFSET(gEffBubble1Tex));
```

Comment:
```suggestion
            gSPSegment(&gfx[1], 0x08, SEGMENTED_TO_VIRTUAL(gEffBubble1Tex));
```

Reply:
Doesn't match because `SEGMENTED_TO_VIRTUAL` uses the `PHYSICAL_TO_VIRTUAL` macro internally while this operation doesn't

Reply:
Oh Right forgot about that, what about 
```suggestion
gSPSegment(&gfx[1], 0x08, VIRTUAL_TO_PHYSICAL(SEGMENTED_TO_VIRTUAL(gEffBubble1Tex);
```
For whatever reason the bubble textures seem to do this.